pkg/controller/storagecluster/placement_test.go

```diff
@@ -13,108 +13,105 @@ import (
+	MatchExpressions: []metav1.LabelSelectorRequirement{},
+}
+var emptyNodeAffinity = corev1.NodeAffinity{
+	RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+		NodeSelectorTerms: []corev1.NodeSelectorTerm{
+			corev1.NodeSelectorTerm{
+				MatchExpressions: []corev1.NodeSelectorRequirement{},
 			},
 		},
 	},
 }
+var emptyPlacements = map[rookv1.KeyType]rookv1.Placement{
+	"all": rookv1.Placement{},
+}
 
 func TestGetPlacement(t *testing.T) {
-	// Case 1: Defaults are preserved i.e no placement and no label selector
-	sc := &ocsv1.StorageCluster{}
-	mockStorageCluster.DeepCopyInto(sc)
-	assert.Equal(t, defaultLabelPlacement["all"], getPlacement(sc, "all"))
-
-	// Case 2: The configured Placements override the defaults
-	sc = &ocsv1.StorageCluster{}
-	mockStorageCluster.DeepCopyInto(sc)
-	sc.Spec.Placement = mockPlacements
-	assert.Equal(t, defaultLabelPlacement["all"], getPlacement(sc, "all"))
-
-	// Case 3: LabelSelector to modify the default placements correctly
-	sc = &ocsv1.StorageCluster{}
-	mockStorageCluster.DeepCopyInto(sc)
-	sc.Spec.LabelSelector = &metav1.LabelSelector{
-		MatchExpressions: []metav1.LabelSelectorRequirement{
-			metav1.LabelSelectorRequirement{
-				Key:      WorkerAffinityKey,
-				Operator: metav1.LabelSelectorOpExists,
+	cases := []struct {
+		placements        rookv1.PlacementSpec
+		labelSelector     *metav1.LabelSelector
+		expectedPlacement rookv1.Placement
+	}{
+		{
+			// Case 1: Defaults are preserved i.e no placement and no label selector
+			placements:    rookv1.PlacementSpec{},
+			labelSelector: nil,
+			expectedPlacement: rookv1.Placement{
+				NodeAffinity: defaults.DefaultNodeAffinity,
+				Tolerations:  defaults.DaemonPlacements["all"].Tolerations,
 			},
 		},
-	}
-	assert.Equal(t, mockPlacements["all"], getPlacement(sc, "all"))
-
-	// Case 4: The LabelSelector modifies the configured Placements correctly
-	sc = &ocsv1.StorageCluster{}
-	mockStorageCluster.DeepCopyInto(sc)
-	sc.Spec.Placement = mockPlacements
-	sc.Spec.LabelSelector = &metav1.LabelSelector{
-		MatchExpressions: []metav1.LabelSelectorRequirement{
-			metav1.LabelSelectorRequirement{
-				Key:      MasterAffinityKey,
-				Operator: metav1.LabelSelectorOpExists,
+		{
+			// Case 2: The configured Placements override the defaults
+			placements:    emptyPlacements,
+			labelSelector: nil,
+			expectedPlacement: rookv1.Placement{
+				NodeAffinity: defaults.DefaultNodeAffinity,
+			},
+		},
+		{
+			// Case 3: LabelSelector to modify the default Placements correctly
+			placements:    rookv1.PlacementSpec{},
+			labelSelector: &workerLabelSelector,
+			expectedPlacement: rookv1.Placement{
+				NodeAffinity: &corev1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &workerNodeSelector,
+				},
+				Tolerations: defaults.DaemonPlacements["all"].Tolerations,
+			},
+		},
+		{
+			// Case 4: The LabelSelector modifies the configured Placements correctly
+			placements:    emptyPlacements,
```

Comment:
Shouldn't we add some real placement instead of empty to test this?
If placement and labelSelector is provided, what is the behaviour?

Reply:
Yes, this should also be covered.

Reply:
I can see the need for that, will add another test case.